src/hooks/useSubmitTransactions.ts

```diff
@@ -1,64 +1,55 @@
-import { useContractWrite } from 'wagmi';
+import { useToast } from '@chakra-ui/react';
 import { useAddRecentTransaction } from '@rainbow-me/rainbowkit';
-import { useToast, ToastId } from '@chakra-ui/react';
-import { useRef } from 'react';
-
+import { useContractWrite } from 'wagmi';
 import { UseContractWriteArgs } from 'wagmi/dist/declarations/src/hooks/contracts/useContractWrite';
+import { formatToastMessage } from '../utils/helpers';
 
-export function useSubmitTransaction(contractConfig: Omit<UseContractWriteArgs, 'addressOrName'>) {
-  const writeConfig = {
-    addressOrName: process.env.REACT_APP_LOCAL_CONTRACT_ADDRESS || '',
-    ...contractConfig,
-  };
+type UseSubmitTransactionsArgs = UseContractWriteArgs & {
+  toastDescription?: string;
+  transactionDescription?: string;
+};
 
-  const pendingToastId = useRef<ToastId | undefined>();
-  const toastText = useRef<string>();
-  const toast = useToast();
+export function useSubmitTransaction(
+  contractConfig: Omit<UseSubmitTransactionsArgs, 'addressOrName'>
+) {
+  // Constants
+  const defaultSuccessToast = 'Transaction submitted';
+  const defaultTransactionDescription = 'Unknown transaction submitted';
+  const toastDuration = 5000;
 
+  const toast = useToast();
   const addRecentTransaction = useAddRecentTransaction();
 
-  const contractWrite = useContractWrite({
+  const { writeAsync } = useContractWrite({
+    addressOrName: process.env.REACT_APP_LOCAL_CONTRACT_ADDRESS || '',
     onSuccess(data) {
```

Comment:
We discussed the use of onSettled and onSuccess. I'll use hardhat to test the differences between the two.